### PR TITLE
keep state file to signal reindex post-compaction

### PR DESCRIPTION
**Pull request by staltz**
This PR solves 3 problems:

- Issue #226
- `reindex` was wrong for bitset indexes
- Keeping track of when should `reindex()` be called post-compact should be a jitdb concern
  - (Currently ssb-db2 persists three state files for this, one of them for jitdb reindexing)

These changes should cause a **breaking change** (because of the added argument in `paginate()`) and we should also remove the ssb-db2 `post-compact-reindex-jit` state file.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1519,6 +1592,8 @@ module.exports = function (log, indexesPath) {
           }
         }
 
+        if (index.bitset) index.bitset.removeRange(seq, Infinity)
```
Comment (staltz) [inline index.js:1595]: This was a whoops

Comment (arj03) [reply index.js:1598]: WOAH

File: index.js [hunk 2]
```diff
@@ -1251,17 +1285,59 @@ module.exports = function (log, indexesPath) {
...
+    )
+  }
+
   function countBitsetSlice(bitset, seq, descending) {
     if (!seq) return bitset.size()
     else return bitset.size() - seq
   }
 
-  function paginate(operation, seq, limit, descending, onlyOffset, sortBy, cb) {
+  function paginate(
```
Comment (arj03) [inline index.js:1317]: The README needs update for this :)

File: index.js [hunk 3]
```diff
@@ -1251,17 +1285,59 @@ module.exports = function (log, indexesPath) {
...
+    operation,
+    seq,
+    limit,
+    descending,
+    onlyOffset,
+    sortBy,
+    latestMsgKey,
```
Comment (arj03) [inline index.js:1324]: Can we call this compactionLatestMsgKey or something like that? To signal that this is only ever used for compaction.

**Comment by arj03** [comment]
end of review. Overall this approach looks good.

**Comment by staltz** [comment]
Thanks for taking a look! I updated it now.

**Review by arj03 — APPROVED** [review]
:star2:

**Pull request merged**